Fix javadoc warnings

```
> Task :jaeger-core:javadoc
/Users/yurishkuro/Uber/jaeger-client-java/jaeger-core/src/main/java/com/uber/jaeger/baggage/BaggageSetter.java:51: warning: no @param for key
  public SpanContext setBaggage(Span span, String key, String value) {
                     ^
/Users/yurishkuro/Uber/jaeger-client-java/jaeger-core/src/main/java/com/uber/jaeger/Configuration.java:257: warning: no @param for statsFactory
  public void setStatsFactor(StatsFactory statsFactory) {
              ^
/Users/yurishkuro/Uber/jaeger-client-java/jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java:177: warning: no @return
  public long getTraceID() {
              ^
/Users/yurishkuro/Uber/jaeger-client-java/jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java:185: warning: no @return
  public long getSpanID() {
              ^
/Users/yurishkuro/Uber/jaeger-client-java/jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java:193: warning: no @return
  public long getParentID() {
              ^
```

more
```
/Users/yurishkuro/Uber/jaeger-client-java/jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientSpanInjectionFilter.java:42: warning: no @param for tracer
  public ClientSpanInjectionFilter(Tracer tracer, TraceContext traceContext) {
         ^
/Users/yurishkuro/Uber/jaeger-client-java/jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientSpanInjectionFilter.java:42: warning: no @param for traceContext
  public ClientSpanInjectionFilter(Tracer tracer, TraceContext traceContext) {
         ^
/Users/yurishkuro/Uber/jaeger-client-java/jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientSpanInjectionFilter.java:42: warning - Tag @link: missing '#': "ClientSpanInjectionFilter(Tracer)"
3
```

--- a/config/checkstyle/suppressions.xml
+++ b/config/checkstyle/suppressions.xml
@@ -5,6 +5,5 @@
 
 <suppressions>
   <suppress checks="AbbreviationAsWordInName"
-    files="SpanContext.java"
-    lines="177,185,193"/>
+    files="SpanContext.java"/>
 </suppressions>

--- a/jaeger-context/src/main/java/com/uber/jaeger/context/Runnable.java
+++ b/jaeger-context/src/main/java/com/uber/jaeger/context/Runnable.java
@@ -16,6 +16,7 @@ package com.uber.jaeger.context;
 
 import io.opentracing.Span;
 
+@SuppressWarnings("JavaLangClash") // this class will be deprecated in the future
 public class Runnable implements java.lang.Runnable {
   private final java.lang.Runnable wrappedRunnable;
   private final TraceContext traceContext;

--- a/jaeger-context/src/test/java/com/uber/jaeger/context/TracedExecutorServiceTest.java
+++ b/jaeger-context/src/test/java/com/uber/jaeger/context/TracedExecutorServiceTest.java
@@ -109,6 +109,7 @@ public class TracedExecutorServiceTest {
   }
 
   @Test
+  @SuppressWarnings("FutureReturnValueIgnored")
   public void testSubmitCallableOfT() {
     tracedExecutorService.submit(wrappedCallable);
 
@@ -119,6 +120,7 @@ public class TracedExecutorServiceTest {
   }
 
   @Test
+  @SuppressWarnings("FutureReturnValueIgnored")
   public void testSubmitRunnable() {
     tracedExecutorService.submit(wrappedRunnable);
 

--- a/jaeger-core/src/main/java/com/uber/jaeger/Configuration.java
+++ b/jaeger-core/src/main/java/com/uber/jaeger/Configuration.java
@@ -251,7 +251,7 @@ public class Configuration {
   }
 
   /**
-   * @param statsFactory the fatory factory
+   * @param statsFactory the factory
    * @deprecated Use {@link #setStatsFactory(StatsFactory)} instead
    */
   @Deprecated

--- a/jaeger-core/src/main/java/com/uber/jaeger/Configuration.java
+++ b/jaeger-core/src/main/java/com/uber/jaeger/Configuration.java
@@ -251,6 +251,7 @@ public class Configuration {
   }
 
   /**
+   * @param statsFactory the fatory factory
    * @deprecated Use {@link #setStatsFactory(StatsFactory)} instead
    */
   @Deprecated

--- a/jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java
+++ b/jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java
@@ -172,6 +172,7 @@ public class SpanContext implements io.opentracing.SpanContext {
 
   /**
    * @deprecated use {@link SpanContext#getTraceId()} instead.
+   * @return same as getTraceId
    */
   @Deprecated
   public long getTraceID() {
@@ -180,6 +181,7 @@ public class SpanContext implements io.opentracing.SpanContext {
 
   /**
    * @deprecated use {@link SpanContext#getSpanID()} instead.
+   * @return same as getSpanId
    */
   @Deprecated
   public long getSpanID() {
@@ -188,6 +190,7 @@ public class SpanContext implements io.opentracing.SpanContext {
 
   /**
    * @deprecated use {@link SpanContext#getParentID()} instead.
+   * @return same as getParentId
    */
   @Deprecated
   public long getParentID() {

--- a/jaeger-core/src/main/java/com/uber/jaeger/baggage/BaggageSetter.java
+++ b/jaeger-core/src/main/java/com/uber/jaeger/baggage/BaggageSetter.java
@@ -45,6 +45,7 @@ public class BaggageSetter {
    * on the {@link Span}.
    *
    * @param  span  the span to set the baggage on
+   * @param  key   the baggage key to set
    * @param  value the baggage value to set
    * @return       the SpanContext with the baggage set
    */

--- a/jaeger-core/src/test/java/com/uber/jaeger/ConfigurationTest.java
+++ b/jaeger-core/src/test/java/com/uber/jaeger/ConfigurationTest.java
@@ -19,6 +19,7 @@ import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
 
 import com.uber.jaeger.Configuration.ReporterConfiguration;
 import com.uber.jaeger.Configuration.SamplerConfiguration;
@@ -174,6 +175,7 @@ public class ConfigurationTest {
     System.setProperty(Configuration.JAEGER_AGENT_PORT, "6832");
     try {
       Configuration.SenderConfiguration.fromEnv().getSender();
+      fail("expecting exception");
     } catch (RuntimeException re) {
       // we need to catch it here instead of using @Test(expected = ...) because the SocketException is
       // wrapped into a runtime exception
@@ -186,7 +188,7 @@ public class ConfigurationTest {
     System.setProperty(Configuration.JAEGER_AGENT_HOST, "jaeger-agent");
     System.setProperty(Configuration.JAEGER_AGENT_PORT, "6832");
     assertEquals("jaeger-agent", Configuration.ReporterConfiguration.fromEnv().getAgentHost());
-    assertEquals(new Integer(6832), Configuration.ReporterConfiguration.fromEnv().getAgentPort());
+    assertEquals(Integer.valueOf(6832), Configuration.ReporterConfiguration.fromEnv().getAgentPort());
   }
 
   @Test
@@ -337,7 +339,7 @@ public class ConfigurationTest {
     }
   }
 
-  private class CustomSender extends HttpSender {
+  private static class CustomSender extends HttpSender {
     private String endpoint;
 
     public CustomSender(String endpoint) {

--- a/jaeger-core/src/test/java/com/uber/jaeger/senders/HttpSenderTest.java
+++ b/jaeger-core/src/test/java/com/uber/jaeger/senders/HttpSenderTest.java
@@ -15,11 +15,13 @@
 package com.uber.jaeger.senders;
 
 import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
 
 import com.uber.jaeger.Configuration;
 import com.uber.jaeger.thriftjava.Process;
 import com.uber.jaeger.thriftjava.Span;
 
+import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.Base64;
 import java.util.List;
@@ -117,6 +119,7 @@ public class HttpSenderTest extends JerseyTest {
 
     try {
       sender.send(new Process("robotrock"), generateSpans());
+      fail("expecting exception");
     } catch (TException te) {
       assertTrue(te.getMessage().contains("response 401"));
     }
@@ -140,7 +143,9 @@ public class HttpSenderTest extends JerseyTest {
         return Response.status(Response.Status.BAD_REQUEST).build();
       }
 
-      String userAndPass = new String(Base64.getDecoder().decode(authHeader.split("\\s+")[1]));
+      String userAndPass = new String(
+          Base64.getDecoder().decode(authHeader.split("\\s+")[1]),
+          StandardCharsets.US_ASCII);
       if (!userAndPass.equals("jdoe:password")) {
         return Response.status(Response.Status.UNAUTHORIZED).build();
       }

--- a/jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientSpanInjectionFilter.java
+++ b/jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientSpanInjectionFilter.java
@@ -36,7 +36,9 @@ public class ClientSpanInjectionFilter implements ClientRequestFilter, ClientRes
   private final Tracer tracer;
 
   /**
-   * @deprecated use {@link ClientSpanInjectionFilter(Tracer)}
+   * @param tracer tracer
+   * @param traceContext trace context
+   * @deprecated use ClientSpanInjectionFilter(Tracer)
    */
   @Deprecated
   public ClientSpanInjectionFilter(Tracer tracer, TraceContext traceContext) {

